extension/src/main/java/eu/maveniverse/maven/mimir/extension3/MimirLifecycleParticipant.java

```diff
@@ -94,10 +95,21 @@ public void afterSessionEnd(MavenSession session) throws MavenExecutionException
                 }
             });
         } catch (Exception e) {
-            throw new MavenExecutionException("Error closing Mimir session", e);
+            if (skipErrorAtEnd()) {
+                logger.warn("Error closing Mimir session", e);
+            } else {
+                throw new MavenExecutionException("Error closing Mimir session", e);
+            }
         }
     }
 
+    private boolean skipErrorAtEnd() {
+        return Optional.ofNullable(
+                        SessionConfig.defaults().build().effectiveProperties().get("mimir.error-at-end.skip"))
```

Comment:
why constructing config from scratch? if it was thrown, it means MimirUtils.mayGetSession returned a "present" session, that has also config. Why not just use that config to figure out in the consumer itself does it needs to throw or not? Instead let it pop out, construct new config from scratch...